src/sd.c

```diff
@@ -203,7 +212,10 @@ uint8_t sd_list(void)
 #else
             pc_fn = fno.fname;
 #endif
-            if (*pc_fn == '.') {
+            if (*pc_fn == '.' && *(pc_fn + 1) == '\0') {
+                continue;
+            }
+            if (*pc_fn == '.' && *(pc_fn + 1) == '.' && *(pc_fn + 2) == '\0') {
```

Comment:
This seems to do the trick. List (or in code below Erase) the hidden files `.<filename>` but not `.` or `..`